### ISSUE TITLE
Make judging rounds settings-based only

### ISSUE THREAD

**Issue by stenington** [Iridescent-CM/technovation-app#2025]
**Is your feature request related to a problem? Please describe.**
Judging rounds are currently configured through a blend of dates and toggles in a way that's hard to explain and understand. We thought we needed that when live and virtual QFs were going to end at different times, but that created difficulties at a program level so we backed away from it.

**Describe the solution you'd like**
It would be nice now for the judging round changes to be **entirely based on admin settings**. Date configuration should continue to exist for display only.

The judging round settings should be expanded to include:

* Off
* Quarterfinals
* Between rounds
* Semifinals
* Finished

Staff will change the configuration at the right time manually, judges should never be prevented from doing something by date alone.

**Comment by stenington** [comment]
@kmorton2017 @ophelieh @adrichey This is my proposal for changing how the judging round setting works so that admins would have full control over the various judging phases. Off, QFs, and SFs would basically be the same as we already have. Between rounds would show that special between rounds page, and Finished would basically be the same as Off but could be used in the future to show a thank you page or something like that.

Does this make sense? Does it seem like a good thing to have? I've done some exploration and I'm pretty confident I can build it in a reasonable timeframe for this season.

**Comment by adrichey** [comment]
I think it makes sense. If we ever want to go down the road of having separate end dates for virtual and live judging, that should be a bigger conversation that will definitely require a more major rework; so I like this solution now that we aren't going down that road for now. I think these states all make sense to me based on what I have seen for our various states and screens that our end users see. And it will be nice knowing there is one source of truth for the season settings.

**Comment by ophelieh** [comment]
Makes perfect sense. 

To have it documented - 
* In between round is when QFs is still turned on and we are waiting on semifinals to start - admins do not have to change any settings. 
* When semifinals are ready to start, an admin will change toggle from QFs to semifinals.

**Comment by stenington** [comment]
@ophelieh Well with the change I'm suggesting, an admin **would** change from *Quarterfinals* to *Between rounds* when you wanted QF judging to end and the between rounds page to display. That way we'll have 100% control over it, and the platform won't surprise us by turning off quarterfinals early because of some mistake we made with date configuration or whatever.

**Comment by ophelieh** [comment]
Oh so we would add another toggle? Am I understanding that right? 

Thank you!

**Comment by stenington** [comment]
Yeah, it would probably look like this: 

![screen shot 2019-03-07 at 3 22 36 pm](https://user-images.githubusercontent.com/55319/53986757-09f52b80-40ed-11e9-8074-d5357f19e089.png)

When QFs should end, we'd switch it to "Between rounds". When SFs should start, we'd switch it to "Semifinals", and when they're done we'd switch it to "Finished".

**Comment by ophelieh** [comment]
Perfect. Thank you for the image. I get it now. :)

**Comment by stenington** [comment]
[Here is some documentation](https://docs.google.com/document/d/1bFN0yf2M0RrpdxBaq4tudNfhcvGetifY3v1gnHvkV20/edit#) I tried writing to explain the judging phases. You can see the version for blended dates and toggles in the doc history.

**Comment by stenington** [comment]
QA:

* open [this document](https://docs.google.com/document/d/1bFN0yf2M0RrpdxBaq4tudNfhcvGetifY3v1gnHvkV20/edit#)
* make sure the expectations laid out in the "Configuration expectations" section are true for each Judging round setting

**Comment by ophelieh** [comment]
@nickiredescent and @kmorton2017 Could you test this on Tuesday or Wednesday?

Thank you!

**Comment by nickiredescent** [comment]
Tested and seems to work. In particular, the "finished stage" is a bit jarring. Functional, but jarring.

Screenshot of the finished stage.
![Screen Shot 2019-03-19 at 2 21 20 PM](https://user-images.githubusercontent.com/45576078/54642839-c8507300-4a52-11e9-8635-487b42c9f9fa.png)

**Comment by stenington** [comment]
@nickiredescent I totally agree, but I didn't see any evidence that we had a nice thank you page or anything last year. It wouldn't be very hard to add that though. @ophelieh Feel free to open a new ticket for adding a final page for after judging if you want one.

**Comment by ophelieh** [comment]
Created ticket #2053. Feel free to edit as needed.
